### PR TITLE
Apply @mjog's patch so that Vue SFC errors are actually displayed

### PR DESCRIPTION
**Pull request by lol768**
See #7395

# ↪️ Pull Request

Applies the proposed patch from @mjog in #7395

## 🚨 Test instructions

- Deliberately cause a Vue compile error in an SFC - e.g. `<template v-for> key should be placed on the <template> tag.`

- Prior to applying the patch, you will get a useless `Error: Got unexpected undefined` message which doesn't tell you what the problem is:

```
Error: Got unexpected undefined
    at $812806c6461f2963$var$nullthrows (/home/adam/Documents/project/node_modules/@parcel/utils/lib/index.js:3485:17)
    at $f02d6a9d30f55938$export$2e2bcd8739ae039 (/home/adam/Documents/project/node_modules/@parcel/utils/lib/index.js:32342:195)
    at $7c213f7c22b64117$var$writeDiagnostic (/home/adam/Documents/project/node_modules/@parcel/reporter-cli/lib/CLIReporter.js:7435:167)
    at $7c213f7c22b64117$export$12358408d9820617 (/home/adam/Documents/project/node_modules/@parcel/reporter-cli/lib/CLIReporter.js:7401:19)
```

- After the patch, observe that a better error is generated (note though that the name of the component is absent)

```
Build failed.

@parcel/transformer-vue: <template v-for> key should be placed on the <template> tag.

  SyntaxError: <template v-for> key should be placed on the <template> tag.
      at createCompilerError (/home/adam/Documents/project/node_modules/@vue/compiler-core/dist/compiler-core.cjs.prod.js:18:19)
      at /home/adam/Documents/project/node_modules/@vue/compiler-core/dist/compiler-core.cjs.prod.js:3767:45
      at Array.some (<anonymous>)
```

## ✔️ PR Todo

- [ ] Added/updated unit tests for this change
- [X] Filled out test instructions (In case there aren't any unit tests)
- [X] Included links to related issues/PRs

### CODE REVIEW

File: packages/core/utils/src/prettyDiagnostic.js
Comment (mischnic) [inline packages/core/utils/src/prettyDiagnostic.js]:
```suggestion
      let code = codeFrame.code;
      if (code == null && options && filePath != null) {
        code = await options.inputFS.readFile(filePath, 'utf8');
      }
```

**Comment by mischnic** [comment]
You also need to remove the line importing `nullthrows`, otherwise the linter will fail.

**Review by mischnic — APPROVED** [review]

**Comment by mischnic** [comment]
Please merge v2 into your branch or [allow edits by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork), otherwise we can't merge this. (And this is the case for your other recent PRs as well)

**Comment by lol768** [comment]
First, thanks for taking the time to review @mischnic 

I'm not sure quite what's going on, but that checkbox doesn't appear for me:

image

I wonder if it's because I forked it to my client's org instead of my own personal account - it does say in the article "you can allow commits on branches you've created from forks owned _by your personal account_"

I've rebased this PR, I'll do the same with the others too now

**Comment by lol768** [comment]
Others are rebased now, please shout if you need me to do it again at any point - it's not a problem :slightly_smiling_face:

**Comment by mischnic** [comment]
> Others are rebased now, please shout if you need me to do it again at any point - it's not a problem 🙂

@lol768  😉

**Comment by lol768** [comment]
Rebased :)

**Comment by lol768** [comment]
Cheers

**Pull request merged**
